lisa/environment.py

```diff
@@ -366,6 +368,14 @@ def _validate_single_default(
             has_default = True
         return has_default
 
+    # Sometimes the nodes are created before the environment log or working directories
+    # are accessed for the first time and therefore the nodes create these directories.
+    # However, the 'create_folder' function will throw an exception if a directory
+    # already exists.
+    def _ensure_dirs_created(self) -> None:
```

Comment:
The log folder should be created in the flow. And the working path should be created explicitly when first using. The working folder is not used in most case, so it won't be created for each run. Can you show me more details?